mattermost/mattermost
# Add post priority to incoming and outgoing webhook

#### Summary
This PR adds the ability to assign priority to incoming webhooks and the responses from outgoing webhooks.

#### Release Note

```release-note
Added post priority to incoming webhooks and outgoing webhook responses
```


### To Test:

#### Incoming:

1. Create an incoming webhook

    - Title: `test`
    - Channel: `Off Topic`

2. Copy the URL provided and use in the below command.
3. Replace `WEBHOOK_URL_HERE` with the webhook url and run in the terminal.

    ```bash
    curl -i -X POST -H 'Content-Type: application/json' -d '{"text": "Hello, this is some text\nThis is more text. 🎉", "priority":{"priority":"urgent","requested_ack":true,"persistent_notifications":false}}' WEBHOOK_URL_HERE
    ```

4. View in Mattermost the requested ack post.

#### Outgoing:

1. Create an outgoing webhook with the below values:

    - Title: `test`
    - Content Type: `application/json`
    - Channel: `Off Topic`
    - Trigger Words: `webhook`
    - Callback URL: `http://localhost:12345`

2. Open a terminal and paste the bash script and press enter to start it.

    ```bash
    #!/bin/bash

    # Define the port to listen on
    PORT=12345

    # Define the JSON payload to respond with
    JSON_PAYLOAD='{"text": "Hello, this is some text. This is more text. 🎉", "priority":{"priority":"urgent","requested_ack":true,"persistent_notifications":false}}'

    # Start a loop that listens for connections and responds with the JSON payload
    while true; do
    echo -e "HTTP/1.1 200 OK\nContent-Type: application/json\n\n$JSON_PAYLOAD" | nc -l "$PORT"
    done
    ```

3. Open a browser to `http://localhost:12345` to confirm it's working.

4. Navigate to the off topic channel and make a post with text contents of `webhook`

5. See response with priority assigned.
